File: mavkit-python/examples/connect_tcp.py
import asyncio
import os

import mavkit


async def main():
    addr = os.environ.get("MAVKIT_EXAMPLE_TCP_ADDR", "127.0.0.1:5760")

    vehicle = await mavkit.Vehicle.connect_tcp(addr)
    identity = vehicle.identity()
    current_mode = vehicle.available_modes().current().latest()
    mode_name = current_mode.name if current_mode is not None else "unknown"

    print(
        f"connected: sys={identity.system_id} comp={identity.component_id} "
        f"autopilot={identity.autopilot} vehicle_type={identity.vehicle_type} "
        f"mode={mode_name}"
    )

    await vehicle.disconnect()


asyncio.run(main())

File: examples/connect_tcp.rs
use mavkit::Vehicle;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let addr =
        std::env::var("MAVKIT_EXAMPLE_TCP_ADDR").unwrap_or_else(|_| "127.0.0.1:5760".to_string());

    let vehicle = Vehicle::connect_tcp(&addr).await?;
    let identity = vehicle.identity();
    let current_mode = vehicle.available_modes().current().wait().await?;
    let armed = vehicle.telemetry().armed().wait().await?.value;

    println!(
        "connected: sys={} comp={} autopilot={:?} vehicle_type={:?} mode={} armed={}",
        identity.system_id,
        identity.component_id,
        identity.autopilot,
        identity.vehicle_type,
        current_mode.name,
        armed,
    );

    vehicle.disconnect().await?;
    Ok(())
}
